EIPS/eip-7867.md

```diff
@@ -0,0 +1,565 @@
+
+###### `200` Confirmed
+
+Status `200` MUST NOT be returned if any calls in the batch failed (including
+batch rollback, and the `onFailure` modes `halt`/`continue`).
+
+###### `207` Partial Success
+
+Status `207` SHALL be returned only when all of the following are true:
+
+* At least one call in the batch has been included and succeeded;
+* At least one call in the batch with an `onFailure` mode of `continue`
+   has been included and failed;
+* No calls with an `onFailure` mode of `rollback` have been included and failed;
+* No calls with an `onFailure` mode of `halt` have been included and failed;
+  and
+* The batch is complete.
+
+###### `500` Chain Rules Failure
+
+To clarify, status `500` is the correct code when the batch has rolled back _or_
+when all calls are non-critical and have all failed.
+
+If any calls are included and succeeded, one of `200`, `207`, or `600` should be
+returned instead.
+
+###### `600` Partial Chain Rules Failure
+
+Status `600` SHALL be returned only when all of the following are true:
+
+* At least one call in the batch has been included and succeeded;
+* At least one call in the batch with an `onFailure` mode of `halt` has been
+  included and failed;
+* No calls with an `onFailure` mode of `rollback` have been included and failed;
+  and
+* The batch is complete.
+
+#### `wallet_getCapabilities`
+
+The response to `wallet_getCapabilities` indicates what call-scope `onFailure`
+modes are supported for each supported batch-scope `atomicity` level for
+batches with two or more calls. Support, here, means "natively supports." A
+wallet that offers `strict` atomicity but not `loose` MUST NOT advertise
+support for `loose` (even if the wallet will upgrade `loose` to `strict`
+without an error.)
+
+The wallet:
+
+* MAY respond with one, two, or three `atomicity` levels.
+* MAY respond with one, two, or three `onFailure` modes in each `atomicity`
+  level. The levels do not need to support the same modes.
+* MUST include the particular atomicity / onFailure combination if it is
+  supported _at all_. For example, if particular orderings are impossible—say
+  `rollback` before `halt` is fine, but `halt` before `rollback` is not—then
+  both `rollback` and `halt` have to be included in the array.
+
+##### Examples
+
+###### Plain Externally Owned Account (EOA)
+
+A plain EOA might offer `halt` functionality by submitting one transaction per
+block, and `continue` by submitting all calls at once.
+
+```js
+{
+    "0x1": {
+        "flowControl": {
+            "none": [ "halt", "continue" ]
+        }
+    }
+}
+```
+
+###### Shielded Mempool Externally Owned Account (EOA)
+
+Unlike a plain EOA, a shielded mempool can provide additional guarantees about
+transaction atomicity. In this example, the wallet does not offer the
+`onFailure` mode of `halt` when using `none` atomicity, but does when using
+`loose`.
+
+```js
+{
+    "0x1": {
+        "flowControl": {
+            "none": [ "rollback", "continue" ]
```

Comment:
how do you rollback in `none`?

Reply:
I believe the options should be `halt` and `continue`. Right @SamWilsn?